java/private/create_jvm_test_suite.bzl

```diff
@@ -91,6 +91,26 @@ def create_jvm_test_suite(
 
     tests = []
 
+    # Optimization for classpath, reduces the duplicate dependencies for each test to instead rely on one target
+    deps_lib_name = "%s-test-deps-lib" % name
+    define_library(
```

Comment:
Not entirely sure but will these two new jars include test resources? We are seeing an issue when using liquibase in a test, as it complains about duplicated resources files from different jars.

Reply:
It will not create any jar as there are no sources to this library. It should only be a bazel construct which allows aspects to simplify the class path resolution.

Reply:
Hmm, that's not what we observe. Yes it doesn't container any Java source code, but it contains test resources. Could it be https://github.com/brian-mcnamara/rules_jvm/blob/d899e276ff1a4a9b98272ed6cd31991f09d7839a/java/private/create_jvm_test_suite.bzl#L16?

Reply:
Does something like https://github.com/bazel-contrib/rules_jvm/compare/main...honnix:rules_jvm:filter-out-resources?expand=1 make sense to you?

Reply:
Taking a look, want to see if data is actually needed at all, or whether we could just remove this all together.

Reply:
Please feel free to ignore my draft and go ahead fixing it. Thank you for looking into this issue. 🙏

Reply:
Sorry for the delay, my environment was being slow yesterday, still looking at it :)

Reply:
@honnix , I was creating a test case and seems like this was the case even before my change: https://github.com/bazel-contrib/rules_jvm/compare/main...brian-mcnamara:rules_jvm:resourceRemoval?expand=1
Both the ["test-lib"](https://github.com/brian-mcnamara/rules_jvm/blob/main/java/private/create_jvm_test_suite.bzl#L80) and actuall [test](https://github.com/brian-mcnamara/rules_jvm/blob/main/java/private/create_jvm_test_suite.bzl#L99) have the resource as part of it. 

Is it the case that your test suites only contain test sources (such that you never get into the logic here: https://github.com/brian-mcnamara/rules_jvm/blob/main/java/private/create_jvm_test_suite.bzl#L71

Reply:
Sorry I'm out of town these days and don't have computer with me. I can look into it in depth tomorrow.

> Is it the case that your test suites only contain test sources (such that you never get into the logic here:

I don't think that is the case.

Reply:
That said, I think it is still not ideal having things duplicated. Maybe resources should not be included in `-test-lib` jar (I don't know the reason why they are explicitly included).

Reply:
Looks like it was like this since the beginning https://github.com/bazel-contrib/rules_jvm/commit/7c008f2e785812ac2bb2898aa47e8082870d39de#diff-7e53ad4b884087fec5fdb0e9b2cf5d2f4fd2390dc68161d0a09878977a34414b

Reply:
Ooph, not great, but thanks for confirming my test was not doing something crazy. I agree it does not make sense to have resources included multiple times. Let me at least propose a PR and see where it goes :)

Reply:
https://github.com/bazel-contrib/rules_jvm/pull/338 created PR, will see what the maintainers think of the approach.